### PR TITLE
[JENKINS-61466] Ignore war/work and war/target on frontend tests

### PR DESCRIPTION
**Pull request by fqueiruga**
See [JENKINS-61466](https://issues.jenkins-ci.org/browse/JENKINS-61466).

### Proposed changelog entries

* Entry 1: Frontend tests no longer run on work directories.
* ...

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] JIRA issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@daniel-beck 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a JIRA issue should exist and be labeled as `lts-candidate`

### CODE REVIEW

File: war/jest.config.json [hunk 1]
```diff
@@ -3,5 +3,6 @@
       "\\.hbs$": "jest-handlebars",
       "\\.js$": "babel-jest"
   },
-  "reporters": ["jest-standard-reporter"]
+  "reporters": ["jest-standard-reporter"],
+  "testPathIgnorePatterns": ["<rootDir>/work", "<rootDir>/target", "<rootDir>/node_modules/"]
```
Comment (daniel-beck) [inline war/jest.config.json]: One with trailing `/`, two without? Does it matter?

Comment (fqueiruga) [reply war/jest.config.json]: Don't think so, I'll edit that anyway

**Review by daniel-beck — APPROVED** [review]
Worked for me when I tried it locally (and broke again when I removed it again).

**Review by Wadeck — APPROVED** [review]
The change seems to be fine.
Is it possible to use a better whitelist instead of a blacklist?
_(we better know where the tests are instead where they are not)_

Totally subjective situation: I often backup my /work into /work2, /work3, etc.

**Comment by daniel-beck** [comment]
> Totally subjective situation: I often backup my /work into /work2, /work3, etc.

`work_old/`, `work_older2/`, … 😦

**Comment by fqueiruga** [comment]
Suppose it can be, I'll look into it

**Review by fcojfernandez — APPROVED** [review]

**Review by varyvol — APPROVED** [review]

**Comment by fqueiruga** [comment]
I've been trying this out and it should work for you @Wadeck , as the blacklist is of a pattern, not a folder.

**Comment by daniel-beck** [comment]
@fqueiruga I think the idea is to update the PR to use a whitelist rather than a blacklist.

**Comment by fqueiruga** [comment]
Can this be squash-merged ?

**Comment by Wadeck** [comment]
With your latest change is the testPathIgnorePatterns still meaningful? (trying to avoid adding new technical debt :p)

**Comment by fqueiruga** [comment]
@Wadeck I don't think it's a problem leaving that in. It could help if we end up adding more folders under test. If you feel like we should remove create a suggestion through here and I'll merge it :)

**Comment by Wadeck** [comment]
@fqueiruga It seems you have deleted the branch and closed this PR by accident.

And yes, if it's not necessary, I prefer to avoid having new useless code in the project. The consequences are more often +confusion +copy/paste than an helpful indication for future potential stuff. I cannot suggest the removal as the branch is gone :)

**Comment by fqueiruga** [comment]
I did remove the branch :| I'll recreate it.

**Pull request closed without merging**